src/queries/upload-queue.tsx

```diff
@@ -0,0 +1,54 @@
+import { createMutation, queryOptions, useQueryClient } from '@tanstack/solid-query'
+import { fetcher } from '~/api'
+import { makeAthenaCall } from '~/api/athena'
+import { parseUploadPath } from '~/utils/parse'
+import { AthenaCallResponse, AthenaOfflineQueueResponse, UploadFilesToUrlsRequest, UploadQueueItem } from '~/types'
+
+const mapOnlineQueueItems = (data: AthenaCallResponse<UploadQueueItem[]>) =>
+  data.result?.map((item) => ({ ...item, ...parseUploadPath(item.url) })).sort((a, b) => b.progress - a.progress) || []
+
+const mapOfflineQueueItems = (data: AthenaOfflineQueueResponse) =>
+  data
+    .filter((item) => item.method === 'uploadFilesToUrls')
+    .flatMap((item) =>
+      (item.params as UploadFilesToUrlsRequest).files_data.map((file) => ({
+        ...file,
+        ...parseUploadPath(file.url),
+        path: file.fn,
+        created_at: 0,
+        current: false,
+        id: '',
+        progress: 0,
+        retry_count: 0,
+      })),
+    )
+
+export const uploadQueue = {
+  prefix: ['upload_queue'],
+
+  online: () => [...uploadQueue.prefix, 'online'],
+  onlineForDongle: (dongleId: string) => [...uploadQueue.online(), dongleId],
+  getOnline: (dongleId: string) =>
+    queryOptions({
+      queryKey: uploadQueue.onlineForDongle(dongleId),
+      queryFn: () => makeAthenaCall<void, UploadQueueItem[]>(dongleId, 'listUploadQueue'),
```

Comment:
I think the athena calls should stay in `src/api`

Reply:
Things like the athena methods available and return types will be in the API though, so you'd still want to export a method like `listUploadQueue` from the API package and use that here